src/workerd/api/node/tests/zlib-nodejs-test.js

```diff
@@ -678,24 +677,23 @@ export const closeAfterError = {
 
 // Tests are taken from:
 // https://github.com/nodejs/node/blob/561bc87c7607208f0d3db6dcd9231efeb48cfe2f/test/parallel/test-zlib-write-after-close.js
-// TODO(soon): Enable the test once `gzip` implementation lands.
-// export const writeAfterClose = {
-//   async test() {
-//     const { promise, resolve } = Promise.withResolvers();
-//     let closeCalled = false;
-//     zlib.gzip('hello', function (err, out) {
-//       const unzip = zlib.createGunzip();
-//       unzip.close(() => (closeCalled = true));
-//       unzip.write('asd', (err) => {
-//         strictEqual(err.code, 'ERR_STREAM_DESTROYED');
-//         strictEqual(err.name, 'Error');
-//         resolve();
-//       });
-//     });
-//     await promise;
-//     assert(closeCalled, 'Close should have been called');
-//   },
-// };
+export const writeAfterClose = {
+  async test() {
+    const { promise, resolve } = Promise.withResolvers();
+    let closeCalled = false;
+    zlib.gzip('hello', function (err, out) {
+      const unzip = zlib.createGunzip();
+      unzip.close(() => (closeCalled = true));
```

Comment:
Can (and should) use `mock.fn()` for this... 
```suggestion
      const closeFn = mock.fn();
      unzip.close(closeFn);
```

Then below you can assert on the number of times `closeFn` was called. Refer to the node.js documentation for mock.fn for more details.